ts/indicators/ichimoku-kinko-hyo.src.ts

```diff
@@ -0,0 +1,1009 @@
+                },
+                senkouSpan: {
+                    styles: {
+                        fill: color(this.color).setOpacity(0.2).get()
+                    }
+                }
+            }, this.options);
+        },
+        toYData: function (point: Highcharts.IKHIndicatorPoint): Array<number> {
+            return [
+                point.tenkanSen,
+                point.kijunSen,
+                point.chikouSpan,
+                point.senkouSpanA,
+                point.senkouSpanB
+            ];
+        },
+        translate: function (
+            this: Highcharts.IKHIndicator
+        ): void {
+            var indicator = this;
+
+            SMA.prototype.translate.apply(indicator);
+
+            indicator.points.forEach(function (
+                point: Highcharts.IKHIndicatorPoint
+            ): void {
+                indicator.pointArrayMap.forEach(function (
+                    value: keyof Highcharts.IKHIndicatorPoint
+                ): void {
+                    if (defined(point[value])) {
+                        (point as any)['plot' + value] =
+                        indicator.yAxis.toPixels(
+                            point[value],
+                            true
+                        );
+
+                        // Add extra parameters for support tooltip in moved
+                        // lines
+                        point.plotY = (point as any)['plot' + value];
+                        point.tooltipPos = [
+                            point.plotX,
+                            (point as any)['plot' + value]
+                        ];
+                        point.isNull = false;
+                    }
+                });
+            });
+        },
+        // One does not simply
+        // Render five linesZ
+        // And an arearange
+        // In just one series..
+        drawGraph: function (
+            this: Highcharts.IKHIndicator
+        ): void {
+
+            var indicator = this,
+                mainLinePoints: Array<Highcharts.IKHIndicatorPoint> =
+                    indicator.points,
+                pointsLength: number = mainLinePoints.length,
+                mainLineOptions: Highcharts.IKHIndicatorOptions =
+                    indicator.options,
+                mainLinePath: (Highcharts.SVGElement|undefined) =
+                    indicator.graph,
+                mainColor = indicator.color,
+                gappedExtend: Highcharts.IKHIndicatorGapExtension = {
+                    options: {
+                        gapSize: mainLineOptions.gapSize
+                    }
+                },
+                pointArrayMapLength: number = indicator.pointArrayMap.length,
+                allIchimokuPoints:
+                Array<Array<Highcharts.IKHIndicatorPoint>> =
+                [[], [], [], [], [], []],
+                ikhMap: Highcharts.Dictionary<
+                Array<Highcharts.IKHIndicatorPoint>
+                > = {
```

Comment:
The 80 char line limit kind of obscures the whole code some places in this file.

Reply:
Please use parentheses for multiline typing (see `ts/README.md`). In that case
```ts
ikhMap: Highcharts.Dictionary<(
    Array<Highcharts.IKHIndicatorPoint>
)> = {
```